src/main/java/seedu/meal360/Parser.java

```diff
@@ -0,0 +1,641 @@
+package seedu.meal360;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+import seedu.meal360.exceptions.IngredientNotFoundException;
+import seedu.meal360.exceptions.InvalidNegativeValueException;
+import seedu.meal360.exceptions.InvalidRecipeNameException;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+public class Parser {
+
+    Ui ui = new Ui();
+
+    public String combineWords(String[] input, int startIndex, int length) {
+        StringBuilder word = new StringBuilder(input[startIndex].trim());
+        for (int i = startIndex + 1; i < length; i++) {
+            word.append(" ").append(input[i].trim());
+        }
+
+        return word.toString();
+    }
+
+    public HashMap<String, Integer> parseIngredientName(String[] command) {
+        HashMap<String, Integer> ingredients = new HashMap<>();
+        int flag = 0;
+        StringBuilder add = null;
+        String ingredientName;
+        int ingredientQuantity;
+        int indexOfEqual;
+
+        for (String s : command) {
+            try {
+                indexOfEqual = s.indexOf("=");
+                if (indexOfEqual == -1) {
+                    if (add != null) {
+                        add.append(" ").append(s);
+                    } else {
+                        add = new StringBuilder(s);
+                    }
+                    flag++;
+
+                } else {
+                    if (flag > 0) {
+                        indexOfEqual = s.indexOf("=");
+                        ingredientName = (s.substring(0, indexOfEqual));
+                        ingredientName = add + " " + ingredientName;
+                        ingredientQuantity = Integer.parseInt(s.substring(indexOfEqual + 1));
+                        flag = 0;
+                        add = null;
+                    } else {
+                        ingredientName = s.substring(0, indexOfEqual);
+                        ingredientQuantity = Integer.parseInt(s.substring(indexOfEqual + 1));
+                    }
+
+                    ingredients.put(ingredientName, ingredientQuantity);
```

Comment:
Method seems to be very nested, perhaps try to reduce the number of nests if it does not severely affect your code logic?